CIScripts/Test/NewContainer.Tests.ps1

```diff
@@ -0,0 +1,127 @@
+Param (
+    [Parameter(Mandatory=$false)] [string] $TestenvConfFile,
+    [Parameter(Mandatory=$false)] [string] $LogDir = "pesterLogs"
+)
+
+. $PSScriptRoot\TestConfigurationUtils.ps1
+. $PSScriptRoot\..\Testenv\Testenv.ps1
+. $PSScriptRoot\..\Testenv\Testbed.ps1
+. $PSScriptRoot\PesterLogger\PesterLogger.ps1
+
+Describe "New-Container" {
+    It "Reports container id when container creation succeeds in first attempt" {
+        Invoke-Command -Session $Session {
+            $DockerThatAlwaysSucceeds = @"
+            Write-Output "{0}"
+            exit 0
+"@ -f $Using:ContainerID
+            Set-Content -Path "docker.ps1" -Value $DockerThatAlwaysSucceeds
+        }
+
+        $NewContainerID = New-Container `
+            -Session $Session `
+            -NetworkName "BestNetwork" `
+            -Name "jolly-lumberjack" | Should -Not -Throw
+        $NewContainerID | Should -Be $ContainerID
+    }
+
+    It "Throws exception when container creation fails" {
+        Invoke-Command -Session $Session {
+            $DockerThatAlwaysFails = @"
+            Write-Error "It's Docker here: This is very very bad!"
+            exit 1
+"@
+            Set-Content -Path "docker.ps1" -Value $DockerThatAlwaysFails
+        }
+
+        New-Container `
+            -Session $Session `
+            -NetworkName "NetworkOfFriends" `
+            -Name "jolly-lumberjack" | Should -Throw
+    }
+
+    It "Reports container id when container creation succeeds in second attempt after failing because of known issue" {
+        Invoke-Command -Session $Session {
+            $DockerThatSucceedsInSecondAttempt = @'
+            if ($args[0] -eq "run") {{
+                $Hope = "HopeForSuccess"
+                if (Test-Path $Hope) {{
+                    Write-Output "{0}"
+                    Remove-Item $Hope
+                    exit 0
+                }} else {{
+                    Set-Content -Path $Hope -Value "New hope"
+                    Write-Output "{0}"
+                    Write-Error "It's Docker here: CreateContainer: failure in a Windows system call. Try again. Good luck!"
+                    exit 1
+                }}
+            }} else {{
+                Write-Output "{0}"
+                exit 0
+            }}
+'@ -f $Using:ContainerID
+            Set-Content -Path "docker.ps1" -Value $DockerThatSucceedsInSecondAttempt
+        }
+
+        $NewContainerID = New-Container `
+            -Session $Session `
+            -NetworkName "SoftwareDefinedNetwork" `
+            -Name "jolly-lumberjack" | Should -Not -Throw
+        $NewContainerID | Should -Be $ContainerID
+    }
+
+    It "Throws exception when container creation fails in first attempt and reports unknown issue" {
+        Invoke-Command -Session $Session {
+            $HopeStorage = "HopeForLuckyTry"
+            Remove-Item $HopeStorage -ErrorAction Ignore
```

Comment:
You're removing this file before the test here, but not in the other test. Why?

Reply:
Maybe use [this](https://github.com/pester/Pester/wiki/TestDrive) for all files created in tests